src/main/java/lesson2/problem9/IntersectingLists.java

```diff
@@ -0,0 +1,38 @@
+package lesson2.problem9;
+
+import lesson2.problem0.Link;
+import lesson2.problem0.LinkedList;
+
+public class IntersectingLists {
+
+    public static Link getListIntersection(Link list1, Link list2) {
+        Link a = list1;
+        Link b = list2;
+
+        while (a != b) {
+            a = a == null ? list2 : a.next;
+            b = b == null ? list1 : b.next;
```

Comment:
I do not understand this logic. You need to determine the length of each list first, compute the difference and move the head pointer of the longer list by that offset. Then you should iterate through on both lists at the same time and whenever they point to the same node, you found the intersection.

Reply:
an explanation from leetcode:

"We can use two iterations to do that. In the first iteration, we will reset the pointer of one linkedlist to the head of another linkedlist after it reaches the tail node. In the second iteration, we will move two pointers until they points to the same node. Our operations in first iteration will help us counteract the difference. So if two linkedlist intersects, the meeting point in second iteration must be the intersection point. If the two linked lists have no intersection at all, then the meeting pointer in second iteration must be the tail node of both lists, which is null"

Reply:
You are correct, I am taking my earlier comment back. 

This is pretty smart and essentially does the same thing if we would have determined the length of the individual lists first and then compute the difference and use that as an offset to set the pointer starting position of the longer list.

This solution basically works as follows. Given list1 and list2 linked lists, let's assume list1 is shorter than list2. So pointer "a" will reach the end of list1 sooner than "b" reaches the end of list2. When "a" reached the end of list1, it will be moved to the head list2, which is the longer list. When "b" reaches the end of list2, "a" will point the element at the offset equal to the length difference between lists and then "b" will point to the head of list1. After that both pointers will be moved at the same time and essentially perform the same logic as our solution. It might be harder to come up with this compact solution though, but it is pretty elegant.